frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -626,14 +626,7 @@ func dispatchRun(d *dispatchState, c *instructions.RunCommand, proxy *llb.ProxyE
 		args = withShell(d.image, args)
 	}
 	env := d.state.Env()
-	opt := []llb.RunOption{llb.Args(args)}
-	for _, arg := range d.buildArgs {
-		if arg.Value != nil {
-			env = append(env, fmt.Sprintf("%s=%s", arg.Key, arg.ValueString()))
-			opt = append(opt, llb.AddEnv(arg.Key, arg.ValueString()))
-		}
-	}
-	opt = append(opt, dfCmd(c))
+	opt := []llb.RunOption{llb.Args(args), dfCmd(c)}
```

Comment:
what about the `commitToHistory()`? It doesn't seem to make sense to put a buildarg value there if it has been actually overridden by env. I'm not sure what the old behavior is (eg. should the arg be ignored?)

Reply:
the ARG is there in the history with the old builder as well. That's why it was added here. If you want to revisit, maybe another PR?

Reply:
```
FROM busybox
ARG foo=arg
ENV foo bar${foo:-_}baz
RUN env | grep foo= && echo foo=$foo
```

```
$ docker history $(DOCKER_BUILDKIT=0 docker build -q --progress plain --no-cache .)
IMAGE               CREATED                  CREATED BY                                      SIZE                COMMENT
61d3d672d952        Less than a second ago   /bin/sh -c env | grep foo= && echo foo=$foo     0B
00980ee6176d        2 seconds ago            /bin/sh -c #(nop)  ENV foo=barargbaz            0B
433321860916        2 seconds ago            /bin/sh -c #(nop)  ARG foo=arg                  0B
e4db68de4ff2        5 weeks ago              /bin/sh -c #(nop)  CMD ["sh"]                   0B
<missing>           5 weeks ago              /bin/sh -c #(nop) ADD file:b265aa0ea2ef7ff1f…   1.22MB
$ docker history $(DOCKER_BUILDKIT=1 docker build -q --progress plain --no-cache .)
IMAGE               CREATED                  CREATED BY                                      SIZE                COMMENT
64d260c4d5ae        Less than a second ago   RUN |1 foo=arg /bin/sh -c env | grep foo= &&…   0B                  buildkit.dockerfile.v0
<missing>           292 years ago            ENV foo=barargbaz                               0B                  buildkit.dockerfile.v0
<missing>           292 years ago            ARG foo=arg                                     0B                  buildkit.dockerfile.v0
<missing>           5 weeks ago              /bin/sh -c #(nop)  CMD ["sh"]                   0B
<missing>           5 weeks ago              /bin/sh -c #(nop) ADD file:b265aa0ea2ef7ff1f…   1.22MB
```

With this PR I got the same:
```
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
566c91a6293a        5 minutes ago       RUN |1 foo=FOO /bin/sh -c env | grep foo= &&…   0B                  buildkit.dockerfile.v0
<missing>           5 minutes ago       ENV foo=barFOObaz                               0B                  buildkit.dockerfile.v0
<missing>           5 minutes ago       ARG foo=arg                                     0B                  buildkit.dockerfile.v0
<missing>           28 hours ago        /bin/sh -c #(nop)  CMD ["sh"]                   0B
<missing>           28 hours ago        /bin/sh -c #(nop) ADD file:9ceca008111a4ddff…   1.22MB
```

Reply:
how is it the same for the old builder? I don't see `1| ..` for that in your output

Reply:
Added a commit to fix this